### ISSUE TITLE
Do you have to call model speedup every time before inference to see performance gains?

### ISSUE THREAD

**Issue by pmmitche** [microsoft/nni#4780]
**Describe the issue**:
I am running a training and pruning process where I train one model for 100 epochs and then prune at various percentages with NNI to create different sparsity models. Each time I create a new pruned model, I use NNI's model speedup before retraining/fine-tuning, and then save the model. However, when I want to time inference for the model's to compare their different inference times, I am seeing very similar times between the unpruned model and the pruned model. However, when I call prune and compress and then call the model speedup _again_ before timing inference, I see better times. I couldn't find this in the documentation, **but really my question is**, do you have to call model speedup every time you load the model and before inference? Or if you call it, and save the model, and then later load it, should the model already be sped up? Any insight would be great, thanks!

**Environment**:
- NNI version: 2.7
- Python version: Python 3.8.10
- PyTorch/TensorFlow version: 1.10.2+cu113
- Is conda/virtualenv/venv used?: Yes

**How to reproduce it?**:
Train a model, prune it, call model speedup, then save the model. Then, load the model and perform inference and compare the time to the unpruned model, after warming up the gpu and averaging inference times over multiple runs.

Then, call the pruner and model speedup again on the already pruned and saved model, and perform inference again to compare.

**Comment by J-shang** [comment]
hello @pmmitche , model speedup will replace the masked modules in your model with compact smaller modules (i.e., using Linear(50, 50) replace Linear(100, 100)). So you are right, save the compact model after call `model speedup`, and then later load it, the model should already be sped up.

**Comment by pmmitche** [comment]
That makes sense, thank you!
